src/collection.rs

```diff
@@ -442,12 +449,53 @@ impl<'a, C: ClientExt> Collection<'a, C> {
     /// Creates a new document from the document given in the body, unless
     /// there is already a document with the _key given. If no _key is given, a
     /// new unique _key is generated automatically.
+    /// Possibly given _id and _rev attributes in the body are always ignored,
+    /// the URL part or the query parameter collection respectively counts.
+    ///
+    /// If the document was created successfully, then the Location header contains
+    /// the path to the newly created document.
+    /// The Etag header field contains the revision of the document.
+    /// Both are only set in the single document case.
+    ///
+    /// If silent is not set to true, the body of the response contains a JSON object with the following attributes:
+    ///
+    /// _id contains the document identifier of the newly created document
+    /// _key contains the document key
+    /// _rev contains the document revision
+    /// If the collection parameter waitForSync is false, then the call returns as soon as the document has been accepted.
+    /// It will not wait until the documents have been synced to disk.
     ///
+    /// Optionally, the query parameter waitForSync can be used to force synchronization of the document creation
+    /// operation to disk even in case that the waitForSync flag had been disabled for the entire collection.
+    /// Thus, the waitForSync query parameter can be used to force synchronization of just this specific operations.
+    /// To use this, set the waitForSync parameter to true. If the waitForSync parameter is not specified or set to false,
+    /// then the collection’s default waitForSync behavior is applied.
+    /// The waitForSync query parameter cannot be used to disable synchronization for collections that have a default waitForSync value of true.
+    ///
+    /// If the query parameter returnNew is true, then, for each generated document, the complete new document is returned under the new attribute in the result.
     /// # Note
     /// this function would make a request to arango server.
     #[maybe_async]
-    pub async fn create_document<T>(&self, _doc: Document<T>) {
-        unimplemented!()
+    pub async fn create_document<T>(
+        &self,
+        _doc: Document<T>,
+        _insert_options: Option<DocumentInsertOptions>,
+    ) -> Result<DocumentResponse<T>, ClientError>
+    where
+        T: Serialize + Debug + DeserializeOwned,
+    {
+        let mut url = self.document_base_url.join("").unwrap();
+        let body = serde_json::to_string(&_doc)?;
+
+        if _insert_options.is_some() {
+            if _insert_options.unwrap().return_new.is_some() {
+                url.query_pairs_mut().append_pair("return_new", "true");
+            }
+        }
+
+        let resp: DocumentResponse<T> =
+            serialize_response(self.session.post(url, body.as_str()).await?.text())?;
```

Comment:
@arn-the-long-beard Here is the line to be blamed for. 

The `serialize_response` is not a wrapper around `serde_json::from_str`. It actually deserailize the response string as `Response<T>` (see [`serialize_response`](https://github.com/fMeow/arangors/blob/d4f0518490fefcc3b1e80644572777e0c3cfc545/src/response.rs#L27)), while the `Response` is actually looking for "error" in json to determine if this request succeeded. 

And the solution is fairly simple. Use `let response: T = serde_json::from_str(text)?` instead of  `serialize_response(text)`.

Reply:
Thank you for your explanation.!! It is working now. 

But I think I still do not understand everything :)

1 - We used serialize_response in other methods in collection and it worked fine, so why is it failing now ?

2 - I see that serialize_response does contain `let response: T = serde_json::from_str(text)?` , so why the same code does fails inside serialize_response but works outside ?

Reply:
`serialize_response` uses `let response: Response<T> = serde_json::from_str(text)?;`. And `Response<T>` is fundamentally different from `T`. 

Here `Response` is a wrapper aims to facilitate conversion from json response to `Result<T, ClientError>`, which is more sensible in rust. When the request errs, a method returns `Err(ClientErr)`, whilist when all things are doing well, a method returns `Ok(T)`. And when deserialize from json, it looks for `"error"` to decide if to deserialize the json to a `Err(ClientError)` or `Ok(T)`. The explanation is a simplied version, the actual implementation uses a middle level data struct to mimic `Result` and then convert to `Result`, because one cannot implement deserialize method for `Result` type due to rust's orphan rule.

Oh, I should name it `deserialize_response`. That's a big typo.

`create_collections` works because the returned json from server are fundamentally different from `create_document`. For methods like create collections, the server returns a json that contains "error":

```json
{"error":false,"code":200,"writeConcern":1,"waitForSync":false,"globallyUniqueId":"h4CC1D1BE3E5B/171","id":"171","cacheEnabled":false,"isSystem":false,"keyOptions":{"allowUserKeys":true,"type":"traditional","lastValue":0},"objectId":"170","name":"test_collection","status":3,"statusString":"loaded","type":2}
``` 

While the json when creating documents are like `{"_id": String("test_collection_create_document/389"), "_key": String("389"), "_rev": String("_at6UzU----")}`. I don't know what would happen if it errs.

Reply:
Haa now I understand. I was wondering if the response from the server was different. I should have look more in the doc.

Thank you a lot for all your explanations !!! it makes a lot of sense now.

For the request to create goes wrong, i'am going to make some test and see how it goes.

Once gain, thank you a lot!!! That is very appreciated. I will rename in deserialize_response :)